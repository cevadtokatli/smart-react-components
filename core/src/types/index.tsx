export type ContentElement = string | JSXChildren

export type JSXChildren = JSX.Element | JSX.Element[]

export interface JSXElementProps {
  [key: string]: any
}

export interface NestedObject<T> {
  [key: string]: T | NestedObject<T>
}

export type PaletteProp = 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark' | (string & {})

export type Partial<T> = {
  [key in keyof T]?: T[key] extends object ? Partial<T[key]> : T[key]
}

export type ResponsiveProp<K extends string, T> = {
  [key in K as K]: T
} & {
  [key in K as `${K}Sm` | `${K}Md` | `${K}Lg` | `${K}Xl`]?: T
}

export type ResponsiveProps<T> =
{ [key in keyof T as key]?: T[key] }
& { [key in keyof T as key extends string ? `${key}Sm` : never]?: T[key] }
& { [key in keyof T as key extends string ? `${key}Md` : never]?: T[key] }
& { [key in keyof T as key extends string ? `${key}Lg` : never]?: T[key] }
& { [key in keyof T as key extends string ? `${key}Xl` : never]?: T[key] }

export interface Size<T> {
  small: T
  medium: T
  large: T
  [key: string]: T
}

export type SizeProp = 'small' | 'medium' | 'large' | (string & {})

export type Value = string | number
