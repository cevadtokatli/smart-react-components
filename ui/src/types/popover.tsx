import { Position } from './'

export interface PopoverCalculationResult {
  arrowStyle: string
  isArrowInHeader: boolean
  overflow: number
  pos: Position
  style: string
}
