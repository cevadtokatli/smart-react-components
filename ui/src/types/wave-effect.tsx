export interface EffectItem {
  element: JSX.Element
  endTime: number
}
