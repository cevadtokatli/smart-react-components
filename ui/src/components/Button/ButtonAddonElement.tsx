import Div from '@smart-react-components/core/Element/Div'
import Svg from '@smart-react-components/core/Element/Svg'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { OrderPosition } from '../../types'
import { toCSSValue } from '../../util/css'
import { getReverseOrderPosition } from '../../util/props'

interface Props extends
  StyledProps,
  ResponsiveProp<'buttonSize', SizeProp> {
  addonPosition: OrderPosition
  hasHover: boolean
  isDisabled: boolean
  isOutline: boolean
  isSeparated: boolean
  isSoft: boolean
  palette: PaletteProp
  shape?: ShapeProp
}

export default styled(Div).attrs<Props>({
  className: 'src-button-addon',
  getButtonSize: (v, t) => `
    font-size: ${t.$.size.button[v].fontSize};
    padding: 0 ${toCSSValue(t.$.size.input[v].padding.x)(v => v / 2)};

    > ${Svg} {
      height: ${t.$.size.button[v].iconSize};
      width: ${t.$.size.button[v].iconSize};
    }
  `,
})<Props>(({ theme, addonPosition, hasHover, isDisabled, isOutline, isSeparated, isSoft, palette, shape }: Props) => `
  align-items: center;
  border: solid 1px;
  border-${OrderPosition[getReverseOrderPosition(addonPosition)].toLowerCase()}: 0;
  border-bottom-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.button[shape]};
  border-top-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.button[shape]};
  box-sizing: border-box;
  display: inline-flex;
  fill: currentcolor;
  flex: 0 0 auto;
  justify-content: center;
  transition: 300ms 0s ease-in-out;
  transition-property: background, border, color;

  ${!isOutline
    ? `
      ${!isSoft
        ? `
          background: ${theme.$.palette[palette].dynamicer};
          border-color: ${theme.$.palette[palette].dynamicer};
          color: ${theme.$.palette[palette].font};

          ${(isSeparated && hasHover)
            ? `
              &:hover {
                background: ${theme.$.palette[palette].dynamicest};
                border-color: ${theme.$.palette[palette].dynamicest};
              }
            `
            : ''
          }
        `
        : `
          background: ${theme.$.palette[palette].softDynamicer};
          border-color: ${theme.$.palette[palette].softDynamicer};
          color: ${theme.$.palette[palette].softFont};

          ${(isSeparated && hasHover)
            ? `
              &:hover {
                background: ${theme.$.palette[palette].softDynamicest};
                border-color: ${theme.$.palette[palette].softDynamicest};
              }
            `
            : ''
          }
        `
      }
    `
    : `
      ${!isSoft
        ? `
          border-color: ${theme.$.palette[palette].dynamicer};
          color: ${theme.$.palette[palette].dynamicer};

          ${(isSeparated && hasHover)
            ? `
              &:hover {
                background: ${theme.$.palette[palette].dynamicer};
                color: ${theme.$.palette[palette].font};
              }
            `
            : ''
          }
        `
        : `
          border-color: ${theme.$.palette[palette].softDynamicer};
          color: ${theme.$.palette[palette].softDynamicer};

          ${(isSeparated && hasHover)
            ? `
              &:hover {
                background: ${theme.$.palette[palette].softDynamicer};
                color: ${theme.$.palette[palette].softFont};
              }
            `
            : ''
          }
        `
      }
    `
  }

  ${isSeparated
    ? `
      cursor: ${!isDisabled ? theme.$.cursor.button : 'default'};
      user-select: none;
    `
    : ''
  }

  ${isDisabled
    ? `
      opacity: ${theme.$.opacity.buttonDisabled};
      pointer-events: none;
    `
    : ''
  }
`)
