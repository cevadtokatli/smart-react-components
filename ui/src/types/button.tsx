import { IntrinsicStyledCoreProps } from '@smart-react-components/core/element-props/intrinsic-styled-core-props'
import { ContentElement, JSXElementProps, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'

export interface ButtonGenericProps<T extends ContentElement> extends
  IntrinsicStyledCoreProps,
  Partial<ResponsiveProp<'size', SizeProp>> {
  children: T | T[]
  elementProps?: JSXElementProps
  hasHover?: boolean
  hasSpace?: boolean
  hasWaveEffect?: boolean
  isActive?: boolean
  isBlock?: boolean
  isDisabled?: boolean
  isFixedSize?: boolean
  isLink?: boolean
  isLoading?: boolean
  isSoft?: boolean
  isOutline?: boolean
  leftAddon?: JSX.Element
  loading?: JSX.Element
  palette?: PaletteProp
  rightAddon?: JSX.Element
  shape?: ShapeProp
  waveEffectPalette?: PaletteProp
}

export interface ButtonListGenericProps extends
  IntrinsicStyledCoreProps,
  Partial<ResponsiveProp<'size', SizeProp>> {
  children: JSX.Element[]
  elementProps?: JSXElementProps
  hasHover?: boolean
  hasWaveEffect?: boolean
  isBlock?: boolean
  isFixedSize?: boolean
  isLink?: boolean
  isSoft?: boolean
  isOutline?: boolean
  palette?: PaletteProp
  shape?: ShapeProp
  waveEffectPalette?: PaletteProp
}

export enum ButtonType {
  BUTTON = 'button',
  RESET = 'reset',
  SUBMIT = 'submit',
}
