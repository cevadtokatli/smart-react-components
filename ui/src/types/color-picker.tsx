import { Color } from '@smart-react-components/core/types'
import { MutableRefObject } from 'react'

export enum ColorPickerFormat {
  HEX,
  RGB,
  HSL,
}

export interface PickerProps {
  color: MutableRefObject<Color>
  onDrag: () => void
  onEnd: (onMove: (e: Event) => void, onEnd: () => void) => void
  onRef?: (v: PickerRef) => void
  onStart: (onMove: (e: Event) => void, onEnd: () => void) => void
}

export interface PickerRef {
  onChange: () => void
  onExternalDrag?: () => void
}
