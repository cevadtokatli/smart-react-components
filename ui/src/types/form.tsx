import { IntrinsicStyledProps } from '@smart-react-components/core/element-props/intrinsic-styled-props'
import { ContentElement, JSXElementProps, PaletteProp, Partial, ResponsiveProp, SetState, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import { FormValue } from '../types'

export interface GenericProps extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  IntrinsicStyledProps {
  active: FormValue | FormValue[]
  children: JSX.Element | JSX.Element[]
  elementProps?: JSXElementProps
  hasBorder?: boolean
  hasHover?: boolean
  hasWaveEffect?: boolean
  isBlock?: boolean
  isDisabled?: boolean
  isRequired?: boolean
  isOutline?: boolean
  isSoft?: boolean
  label?: ContentElement
  leftAddon?: ContentElement
  leftIcon?: ContentElement
  palette?: PaletteProp
  rightAddon?: ContentElement
  rightIcon?: ContentElement
  shape?: ShapeProp
  setActive: SetState<FormValue | FormValue[]>
  waveEffectPalette?: PaletteProp
}
