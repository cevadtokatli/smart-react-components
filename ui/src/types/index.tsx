import { Value } from '@smart-react-components/core/types'

export type FormValue = Value | boolean

export enum InputType {
  EMAIL = 'email',
  PASSWORD = 'password',
  SEARCH = 'search',
  TEL = 'tel',
  TEXT = 'text',
  URL = 'url',
}

export enum OrderPosition {
  LEFT = 1,
  RIGHT = 2,
}

export enum Position {
  LEFT = 1,
  RIGHT = 2,
  TOP = 4,
  BOTTOM = 8,
}

export enum TriggerInteraction {
  NONE = 0,
  CLICK = 1,
  HOVER = 2,
  RIGHT_CLICK = 4,
}
