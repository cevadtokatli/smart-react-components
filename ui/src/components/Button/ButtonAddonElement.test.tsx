import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import ButtonAddonElement from './ButtonAddonElement'
import { OrderPosition } from '../../types'

describe('<ButtonAddonElement />', () => {
  let props

  beforeAll(() => {
    props = {
      addonPosition: OrderPosition.LEFT,
      buttonSize: 'medium',
      hasHover: true,
      isDisabled: false,
      isOutline: false,
      isSeparated: false,
      isSoft: false,
      palette: 'primary',
      shape: 'rectangle',
    }
  })

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component disabled', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isDisabled />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component outline', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isOutline />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component with soft colors', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isSoft />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component outline with soft colors', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isOutline isSoft />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should resize component responsively', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} buttonSize="small" buttonSizeMd="medium" buttonSizeLg="large" />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component at right', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} addonPosition={OrderPosition.RIGHT} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be disabled', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isDisabled />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be separated', () => {
    const { asFragment } = render(wrapTheme(<ButtonAddonElement {...props} isSeparated />))
    expect(asFragment()).toMatchSnapshot()
  })
})
