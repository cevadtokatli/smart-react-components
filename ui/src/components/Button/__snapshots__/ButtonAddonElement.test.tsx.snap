// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonAddonElement /> should be disabled 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
  opacity: 0.7;
  pointer-events: none;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should be separated 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1:hover {
  background: #0552A9;
  border-color: #0552A9;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component at right 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-left: 0;
  border-bottom-right-radius: .125rem;
  border-top-right-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component disabled 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
  opacity: 0.7;
  pointer-events: none;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component outline 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  border-color: #0556B2;
  color: #0556B2;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component outline with soft colors 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  border-color: #a7c9ee;
  color: #a7c9ee;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should render component with soft colors 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #a7c9ee;
  border-color: #a7c9ee;
  color: #033c7c;
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<ButtonAddonElement /> should resize component responsively 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .75rem;
  padding: 0 0.4rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px;
  border-right: 0;
  border-bottom-left-radius: .125rem;
  border-top-left-radius: .125rem;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 300ms 0s ease-in-out;
  transition: 300ms 0s ease-in-out;
  -webkit-transition-property: background,border,color;
  transition-property: background,border,color;
  background: #0556B2;
  border-color: #0556B2;
  color: #fff;
}

@media (min-width:768px) {
  .c0 {
    font-size: .875rem;
    padding: 0 0.45rem;
  }
}

@media (min-width:992px) {
  .c0 {
    font-size: 1rem;
    padding: 0 0.5rem;
  }
}

<div
    class="c0 c1 src-button-addon"
    shape="rectangle"
  />
</DocumentFragment>
`;
